packages/core/src/components/DataTable/index.tsx

```diff
@@ -140,10 +170,24 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
   }
 
   private getTableHeight = (expandedDataList: ExpandedRow[]): number => {
-    const { height, rowHeight, showAllRows } = this.props;
+    const { height, rowHeight, showAllRows, dynamicRowHeight } = this.props;
+    // @ts-ignore _rowHeightCache is missing from DataTable types
+    // eslint-disable-next-line no-underscore-dangle
+    const rowHeights: { [key: number]: number } = this.cache._rowHeightCache;
 
     if (showAllRows) {
-      return expandedDataList.length * getHeight(rowHeight) + this.getColumnHeaderHeight();
+      if (dynamicRowHeight) {
+        if (Object.values(rowHeights).length > 0) {
+          const totalHeight = Object.values(rowHeights).reduce(
+            (sum: number, measuredRowHeight: number) => sum + measuredRowHeight,
+            0,
+          );
+          return totalHeight + this.getColumnHeaderHeight();
+        }
+        this.setState(prevState => ({ ...prevState }));
```

Comment:
`this.forceUpdate()`?

Reply:
Yeah, wasn't sure which one was worse, changed them all over.